### ISSUE TITLE
Xamarin.iOS Problems with Ef Core ChangeTracking: OriginalValue differs directly after db select

### ISSUE THREAD

**Issue by sebastianfilke** [xamarin/xamarin-macios#5174]
Here you can see the information to this problem: aspnet/EntityFrameworkCore#13999

**Comment by spouliot** [comment]
If it happens on device it's likely an AOT issue or a linker issue. You can check this by:

1. Enabling the managed linker in your simulator build (it's disabled by default). If it fails then it's a linker issue;

2. Disabling the managed linker in your device builds (enabled by default). If it works then it's a linker issue;

If the results do not change then it's more likely an AOT issue. Hopefully this can unblock you while this is being investigated.

There's not much knowledge of EF inside the iOS team. Can you provide us with a self-contained test case that we can run and see the issue ? Thanks!

**Comment by sebastianfilke** [comment]
Looks like it's an AOT issue. 
I changed the Linker Behavior to "Don't Link" and "Link Framework SDKs Only" for simulator build (iPhoneSimulator) and device build (iPhone).

Simulator works for both. Device does not work for both cases. It's the same like before.

I changed it here. Is this the right option?
![image](https://user-images.githubusercontent.com/10894484/49017628-96588f80-f189-11e8-89da-e967f416b90a.png)
